### PR TITLE
Fix - Avoid method name collision due to different cases

### PR DESCRIPTION
**Pull request by dsavina**
Fix for #237

### CODE REVIEW

**Comment by moufmouf** [comment]
The PHPStan error here is just to say that an error that was previously ignored in `phpstan.neon` is no longer occurring. We can simply remove the ignored line from phpstan.neon and this should build.

File: src/Utils/BeanDescriptor.php [hunk 1]
```diff
@@ -465,7 +466,7 @@ private function getPivotTableDescriptors(): array
      */
     private function checkForDuplicate(MethodDescriptorInterface $descriptor): void
     {
-        $name = $descriptor->getName();
+        $name = mb_strtolower($descriptor->getName());
```
Comment (moufmouf) [inline src/Utils/BeanDescriptor.php:469]: I think we can safely use `strtolower`.
We are dealing with method names here and PHP internals is case sensitive in the "old fashioned" way. I believe PHP internals does not handle UTF8 so it is internally putting method names in lower case using `strtolower`.

**Review by moufmouf — CHANGES_REQUESTED** [review]

File: src/Utils/BeanDescriptor.php (hunk 1 above)
Comment (dsavina) [reply src/Utils/BeanDescriptor.php:469]: I just tested that:
```php
function é() {}
function É() {}
```
No error, you are perfectly right, I'll change that to `strtolower`!

**Comment by dsavina** [comment]
Requested changes have been made, CI breaks for Oracle DB due to added test.

**Pull request merged**